.github/workflows/release.yaml

```diff
@@ -85,151 +211,201 @@ jobs:
-      if: github.event_name == 'push' || github.event_name == 'workflow_dispatch'
-      run: echo "NUCLIO_LABEL=unstable" >> $GITHUB_ENV
-
-    - name: Set NUCLIO_LABEL to release tag
-      if: github.event_name == 'release'
-      run: echo "NUCLIO_LABEL=${{ steps.release_info.outputs.REF_TAG }}" >> $GITHUB_ENV
-
-    - uses: actions/checkout@v4
-
-    - uses: actions/setup-go@v5
-      with:
-        cache: true
-        go-version-file: go.mod
-
-    - name: Install QEMU
-      if: matrix.arch != 'amd64'
-      uses: docker/setup-qemu-action@v3
-      with:
-        platforms: ${{ matrix.arch }}
-
-    - name: Login to registry
-      uses: docker/login-action@v3
-      with:
-        registry: ${{ env.REPO }}
-        username: ${{ secrets.QUAYIO_DOCKER_USERNAME }}
-        password: ${{ secrets.QUAYIO_DOCKER_PASSWORD }}
-
-    - name: Login to cache registry
-      uses: docker/login-action@v3
-      with:
-        registry: ${{ env.CACHE_REPO }}
-        username: ${{ env.CACHE_REPO_NAME }}
-        password: ${{ secrets.GITHUB_TOKEN }}
-
-    - name: Build ${{ matrix.docker_image_rule }} image
-      run: |
-        make pull-docker-images-cache || true
-        make docker-images
-      env:
-        DOCKER_DEFAULT_PLATFORM: linux/${{ matrix.arch }}
-        DOCKER_IMAGES_RULES: ${{ matrix.docker_image_rule }}
-
-        # this only applies when target is `handler-builder-golang-onbuild`
-        # not applied when target is `handler-builder-golang-onbuild-alpine`
-        SKIP_BUILD_GOLANG_ONBUILD_ALPINE: true
-
-    - name: Push cache images
-      if: env.NUCLIO_LABEL == 'unstable'
-      run: make push-docker-images-cache
-      env:
-        DOCKER_IMAGES_RULES: ${{ matrix.docker_image_rule }}
-        SKIP_BUILD_GOLANG_ONBUILD_ALPINE: true
-
-    - name: Push images
-      run: make push-docker-images
-      env:
-        DOCKER_IMAGES_RULES: ${{ matrix.docker_image_rule }}
-        SKIP_BUILD_GOLANG_ONBUILD_ALPINE: true
-
-    - name: Tag and push stable images
-      if: env.NUCLIO_LABEL != 'unstable' && github.event.release.target_commitish == 'master' && contains(matrix.docker_image_rule, 'dashboard')
-      run: |
-        docker tag "$NUCLIO_DOCKER_REPO/dashboard:$NUCLIO_LABEL-$NUCLIO_ARCH" "$NUCLIO_DOCKER_REPO/dashboard:stable-$NUCLIO_ARCH"
-        docker push "$NUCLIO_DOCKER_REPO/dashboard:stable-$NUCLIO_ARCH"
-      env:
-        DOCKER_IMAGES_RULES: ${{ matrix.docker_image_rule }}
+      - name: Prepare envs
+        run: |
+          echo "NUCLIO_DOCKER_REPO=${{ env.REPO }}/${{ env.REPO_NAME }}" >> $GITHUB_ENV
+          echo "NUCLIO_CACHE_REPO=${{ env.CACHE_REPO }}/${{ env.CACHE_REPO_NAME }}" >> $GITHUB_ENV
+          echo "NUCLIO_ARCH=${{ matrix.arch }}" >> $GITHUB_ENV
+
+      - name: Prepare outputs
+        id: release_info
+        run: |
+          echo "REF_BRANCH=${GITHUB_REF#refs/heads/}" >> $GITHUB_OUTPUT
+          echo "REF_TAG=${GITHUB_REF#refs/tags/}" >> $GITHUB_OUTPUT
```

Comment:
does this still yield the right release tag, when created in the `Create GitHub Release` step above?

Reply:
will use `${{ needs.prepare-inputs.outputs.version }}"` instead and get rid of this step completely